.gitignore

```diff
@@ -152,3 +152,6 @@ result-*
 
 # Mac OS
 .DS_Store
+
+# direnv
+.direnv
```

Comment:
What's this for? I didn't need it.

For context, I've been using

```
echo .envrc >>.git/info/exclude
```

to have a private direnv setup that's ignored by git.

It's not `use flake`, but a custom one though, with special support for `XDG_DATA_DIRS`.
Is that variable handled correctly by `use flake`? It should combine the host and project paths.

Reply:
.direnv is probably created when nix-direnv is used, which puts gcroots for the devshell in that directory.

Yeah I've recently noticed that some people prefer to keep their direnv setup private in order to customize it.
But we can have the best of both worlds with this:

```
use flake
watch_file .envrc.local
[[ -f .envrc.local ]] && source_env .envrc.local
```
I updated the PR.

Generally there doesn't seem to be a clear line between what configuration should be private and what public. Personally I believe it's good to have as much as possible upstream, as long as it doesn't disturb.

So feel free to merge this or close it. I don't have strong feelings about it.

Reply:
The possibility of optimizing .envrc by adding the correct watch_file flags and other checks also seems to be a good reason to have it shared upstream

Reply:
In principle I'm in favor, but I actually have two more issues with the direnv approach, which is that it doesn't make the phase functions available. I've quietly tolerated that in my own use, but combining direnv and an actual dev _shell_ may well lead to bad interactions. For instance it's duplicating `$PATH`. Maybe some other behaviors aren't idempotent in effect either.

The other issues is that we have multiple shells and we need them. Usually I run the `native-clangStdenvPackages` one, for the `clangd` support. How do we do that with direnv?

Reply:
nix-direnv handles XDG_DATA_DIRS correctly: https://github.com/nix-community/nix-direnv/blob/36230478c17bfe2c3fc8d4adbc2e0cdf8ac35aea/direnvrc#L163

Reply:
I just did `nix develop .#nix-clangStdenv` inside nix-direnv and it seems to work as expected, I also not aware of an instance where this was an issue. It's also possible to run `direnv deny` to unload direnv before doing that.

Reply:
There is no way to tell what shell direnv is called from, so `.envrc.shellName.local` doesn't work as far as I know but I also don't know why this is needed in this case.

Reply:
The point is that we won't have to manually invoke `nix develop` anymore, right? So it should be a sufficiently complete alternative for actual productive development. I'm sure that it can be that with a little bit of work.

> There is no way to tell what shell direnv is called from, so `.envrc.shellName.local` doesn't work

I wasn't clear. My idea behind this file is that it would specify the `devShells` subattribute to load; not a specific unix shell implementation like fish or something.

> I just did `nix develop .#nix-clangStdenv` inside nix-direnv and it seems to work as expected, I also not aware of an instance where this was an issue.

It's duplicating `$PATH`. This might lead to nasty surprises when you're changing the build inputs.

If we don't need to call `nix develop`, we don't need to nest shells, and then probably this issue is not a big deal.